src/SwiftTaskProvider.ts

```diff
@@ -128,26 +128,31 @@ export function createBuildAllTask(folderContext: FolderContext): vscode.Task {
  * @returns Build All Task
  */
 export async function getBuildAllTask(folderContext: FolderContext): Promise<vscode.Task> {
-    const tasks = await vscode.tasks.fetchTasks({ type: "swift" });
-
     let buildTaskName = SwiftTaskProvider.buildAllName;
     if (folderContext.relativePath.length > 0) {
         buildTaskName += ` (${folderContext.relativePath})`;
     }
 
     // search for build all task in task.json first
-    let task = tasks.find(
-        task =>
-            task.name === `swift: ${buildTaskName}` &&
-            (task.execution as vscode.ProcessExecution).options?.cwd ===
-                folderContext.folder.fsPath &&
-            task.source === "Workspace"
-    );
+    const workspaceTasks = await vscode.tasks.fetchTasks();
+    let task = workspaceTasks.find(task => {
+        if (task.source !== "Workspace" || task.name !== `swift: ${buildTaskName}`) {
```

Comment:
It looks like that won't work if our default build task is:

```json
		{
			"type": "cmake",
			"label": "App Build",
			"command": "build",
			"targets": [
				"all"
			],
			"problemMatcher": "$swiftc",
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"detail": "Build App for selected configuration",
			"dependsOn": "Configure"
		},
```

Reply:
What, because the label is different?

Reply:
@tristanlabelle ?

Reply:
@adam-fowler: @tristanlabelle is out for the week but yeah that's what I think he meant.
Is it not correct to search for a first kind:build/isDefault:true task?

Reply:
As I mentioned in the related issue, workspaces can have multiple packages in them. The default build task for a workspace might not compile the correct package. But I was thinking about and I can probably compare working directories as well to ensure a default task is the correct task for a package

Reply:
Maybe this change will work

Reply:
Sorry for the wait! I think the latest will work. Thanks!